lua/neoclip/telescope.lua

```diff
@@ -111,6 +120,7 @@ local function entry_maker(entry)
         regtype = entry.regtype,
         filetype = entry.filetype,
         ordinal = table.concat(entry.contents, '\n'),
+        db_entry = entry,
```

Comment:
what's this used for?

Reply:
This is because the new `require("neoclip.storage").set_as_most_recent` function requires a reference to the entry that'll be updated, and it has to be the entire entry, so it can correctly generate a hash for it.

The entry maker function does pass down most of the important things, but not a reference to the entry that's used inside Neoclip' s storage. Therefore, if I wanted to avoid this new db_entry property, I would've had to reconstruct the entry based on the `regtype`, `filetype` and `contents` properties. Something like this:

```lua
local db_entry = {
  regtype = telescope_entry.regtype,
  filetype = telescope_entry.filetype,
  contents = telescope_entry.contents,
}
```

The main problem with this approach is that if you ever change the contents inside the sorted list entries, this would break. And it's also annoyingly verbose.

Sorry for the long explanation, but I like being thorough.

Reply:
I see, but since the hash is only computed from the `regtype` and `contents` fields which the `entry` already has couldn't you just pass that one down? Would then be a bit cleaner I think.

Reply:
Ah, I see. I had the impression the hash required the entire object. I'll make this change too